libs/community/langchain_community/utilities/notdiamond.py

```diff
@@ -0,0 +1,284 @@
+import os
+from importlib import metadata
+from typing import Any, AsyncIterator, Dict, Iterator, List, Sequence
+
+from langchain.chat_models.base import init_chat_model
+from langchain_core.language_models import LanguageModelInput
+from langchain_core.messages.utils import convert_to_messages
+from langchain_core.prompt_values import ChatPromptValue, PromptValue, StringPromptValue
+from langchain_core.runnables import Runnable, RunnableConfig
+
+from langchain_community.adapters.openai import convert_message_to_dict
+
+_USER_AGENT = f"langchain-community/{metadata.version('notdiamond')}"
+_LANGCHAIN_PROVIDERS = {
+    "openai",
+    "anthropic",
+    "google",
+    "mistral",
+    "togetherai",
+    "cohere",
+}
+
+
+class NotDiamondRunnable(Runnable[LanguageModelInput, str]):
+    """
+    See Runnable docs for details
+    https://python.langchain.com/v0.1/docs/expression_language/interface/
+    """
+    import notdiamond as nd
```

Comment:
should be imported within methods that use nd

Reply:
Agreed. Note that I've also imported the library within class scope because of type annotations. If possible I'd like to preserve these, but what do you think?

Reply:
Have actually had to remove this import to address some test and lint failures.